sql/tztime.cc

```diff
@@ -2832,17 +2821,16 @@ main(int argc, char **argv)
     free_root(&tz_storage, MYF(0));
   }
 
-  if(!opt_skip_write_binlog)
-    // Fall back to Aria
-    printf("\\d |\n"
-      "IF (select count(*) from information_schema.global_variables where\n"
-      "variable_name='wsrep_on' and variable_value='ON') = 1 THEN\n"
-      "ALTER TABLE time_zone ENGINE=Aria;\n"
-      "ALTER TABLE time_zone_name ENGINE=Aria;\n"
-      "ALTER TABLE time_zone_transition ENGINE=Aria, ORDER BY Time_zone_id, Transition_time;\n"
-      "ALTER TABLE time_zone_transition_type ENGINE=Aria, ORDER BY Time_zone_id, Transition_type_id;\n"
-      "END IF|\n"
-      "\\d ;\n");
+  if(opt_skip_write_binlog)
+    printf("SET SESSION SQL_LOG_BIN=@save_sql_log_bin;\n"
+           "execute immediate if(@wsrep_is_on, 'SET SESSION WSREP_ON=@save_wsrep_on', 'do 0');\n");
+  else
+    // Change back to Aria
+    printf(
+      "execute immediate if(@wsrep_cannot_replicate_tz, 'ALTER TABLE time_zone ENGINE=Aria', 'do 0');\n"
```

Comment:
Hmm, I see it was like this before, but are the `time_zone` tables always initially guaranteed to be `Aria`  (now and in future releases)? Would it make sense to save their current engine at script start, and revert to it here? (using the same pattern as `@save_sql_log_bin`)?

Reply:
Implemented.